docs/glossary.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: default
+title: lakeFS Glossary
+description: Glossary of all terms related to lakeFS technical internals and the architecture
+nav_order: 52
+has_children: false
+---
+
+# Glossary
+
+### Auditing
+Data auditing is data assessment to ensure its accuracy, security, and efficacy for specific usage. It also involves assessing data quality through its lifecycle and understanding the impact of poor quality data on the organization's performance and revenue. Ensuring data reproducibility, auditability, and governance is one of the key concerns of data engineers today.
+
+### Branch
```

Comment:
I would borrow from the definitions we have on this doc page: https://docs.lakefs.io/understand/branching-model.html. They do a good job explaining from the user's point of view how they should use branches/commits/etc.

Reply:
Updated the definition to include better explanation.